### PR TITLE
Refine mask

### PR DESCRIPTION
**Pull request by dabingnn**
what is done in this pull request.
- refactor `mask. _rebuildSceneGraph` to `base-node.replaceSgNode`
- add listener `anchor-changed`
- fix color and opacity bug when mask exist.

### CODE REVIEW

**Comment by dabingnn** [comment]
@pandamicro @jareguo

File: cocos2d/core/utils/base-node.js [hunk 1]
```diff
@@ -948,6 +948,50 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
         return null;
     },
 
+    /**replace sgNode*/
+
+    replaceSgNode: function(sgNode) {
```
Comment (jareguo) [inline cocos2d/core/utils/base-node.js]: 这个方法建议做成私有的（否则要写 firedoc）

**Comment by jareguo** [comment]
会影响到渲染顺序，先合并吧，有问题再提

**Comment by nantas** [comment]
fireball-x/fireball#434

**Pull request merged**